core/events/tests/test_classic_integration.py

```diff
@@ -280,3 +284,258 @@ def test_texism_titles(self):
+                .get(self.submission.submission_id)
+            db_submission.status = db_submission.PROCESSING
+            session.add(db_submission)
+            session.commit()
+
+            # Submission state should reflect scheduled status.
+            submission, _ = events.load(self.submission.submission_id)
+            self.assertEqual(submission.status, submission.SCHEDULED,
+                             "Submission should have scheduled status.")
+
+    def test_scheduled_status_is_reflected_processing_submission(self):
+        """The submission has been scheduled for publication today."""
+        with self.app.app_context():
+            session = classic.current_session()
+
+            # Publication agent publishes the paper.
+            db_submission = session.query(classic.models.Submission)\
+                .get(self.submission.submission_id)
+            db_submission.status = db_submission.PROCESSING_SUBMISSION
+            session.add(db_submission)
+            session.commit()
+
+            # Submission state should reflect scheduled status.
+            submission, _ = events.load(self.submission.submission_id)
+            self.assertEqual(submission.status, submission.SCHEDULED,
+                             "Submission should have scheduled status.")
+
+    def test_scheduled_status_is_reflected_prior_to_announcement(self):
+        """The submission is being published; not yet announced."""
+        with self.app.app_context():
+            session = classic.current_session()
+
+            # Publication agent publishes the paper.
+            db_submission = session.query(classic.models.Submission)\
+                .get(self.submission.submission_id)
+            db_submission.status = db_submission.NEEDS_EMAIL
+            session.add(db_submission)
+            session.commit()
+
+            # Submission state should reflect scheduled status.
+            submission, _ = events.load(self.submission.submission_id)
+            self.assertEqual(submission.status, submission.SCHEDULED,
+                             "Submission should have scheduled status.")
+
+    def test_scheduled_tomorrow_status_is_reflected(self):
+        """The submission has been scheduled for publication tomorrow."""
+        with self.app.app_context():
+            session = classic.current_session()
+
+            # Publication agent publishes the paper.
+            db_submission = session.query(classic.models.Submission)\
+                .get(self.submission.submission_id)
+            db_submission.status = db_submission.NEXT_DAY
+            session.add(db_submission)
+            session.commit()
+
+            # Submission state should reflect scheduled status.
+            submission, _ = events.load(self.submission.submission_id)
+            self.assertEqual(submission.status, submission.SCHEDULED,
+                             "Submission should be scheduled for tomorrow.")
+
+    def test_publication_failed(self):
```

Comment:
`ERROR_STATE` happens to be extremely rare submission state but it's good this test exists!